cmd/kubeadm/app/componentconfigs/configset.go

```diff
@@ -345,6 +345,5 @@ func Validate(clusterCfg *kubeadmapi.ClusterConfiguration) field.ErrorList {
 		groups = append(groups, group)
 	}
 	sort.Strings(groups) // The sort is needed to make the output predictable
-	klog.Warningf("WARNING: kubeadm cannot validate component configs for API groups %v", groups)
 	return field.ErrorList{}
```

Comment:
@ZhiFeng1993 , can you remove everything else from this func but this line?
Also, the "Currently it prints a warining ..." comment above the function needs removal too.

Reply:
Done. I'm a bit confused about this function though. What's the point of returning an empty list?

Reply:
we spoke briefly about this function during yesterday's kubeadm office hours meeting.

there are two options:
- remove the function until we need it.
- add a TODO on top of the function to identify this is reserved for component config validation in the future.

@rosti should be able to provide more information.

Reply:
@ZhiFeng1993 
> Done. I'm a bit confused about this function though. What's the point of returning an empty list?

An older version of this function was performing actual validation of component configs in the previous implementation of the package. In the new implementation this becomes impossible due to the fact that we can no longer import the validation logic of each component.
As a drop-in replacement, I have started an effort to introduce a standard command line interface for each Kubernetes component to validate its config when invoked from the CLI.
Hence, when I re-implemented the `componentconfigs` package, this function's old body was deleted and replaced with a "warning" placeholder. However, validating the configs is not so critical, plus the effort for the standard CLI for validation has ground to a stop ATM.
So this function can either be deleted or left empty with a TODO. I don't have a personal preference actually.

Reply:
Thank you both for the explanation.  Since you guys have future plans for this function, I'll do as suggested.